### PR TITLE
update is2024_dsu_challenge asr scoring script, bitrate calculation

### PR DESCRIPTION
**Pull request by simpleoier**
## What?

Add a local/score.sh for calculating the average CER of LibriSpeech eval sets, and generate a json file.
Update README with instructions of computing bitrate.
Update the `pyscripts/utils/convert_token2json.py` to support streaming audio input.

### CODE REVIEW

File: egs2/interspeech2024_dsu_challenge/asr2/README.md [hunk 1]
```diff
@@ -6,6 +6,47 @@ This can be referred to the ML-SUPERB [recipe](https://github.com/espnet/espnet/
 
 After download the dataset, please set the `MLSUPERB` to the data directory. The preparation will be automatically done in scripts for each tasks.
 
+# Bitrate calculation
```
Comment (sw005320) [inline egs2/interspeech2024_dsu_challenge/asr2/README.md:9]: Please add the actual bitrate result to the baseline

File: egs2/interspeech2024_dsu_challenge/asr2/local/score.sh [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        "asr_eval3": {
+            "Bitrate": XX.XX
+        }
+    }
+}
+EOF
+
+    done
+
+    echo "Please compute the Bitrate information manually by following the README.md."
```
Comment (sw005320) [inline egs2/interspeech2024_dsu_challenge/asr2/local/score.sh:121]: Is it difficult to integrate it?
If it takes time, we can leave it for now, but it is better to run the bitrate calculation at the same time.

Comment (simpleoier) [reply egs2/interspeech2024_dsu_challenge/asr2/local/score.sh:121]: Sorry, I tried to do it yesterday. It was not that straightforward to integrate it.
Maybe I was wrong. Let me do it now.

Comment (sw005320) [reply egs2/interspeech2024_dsu_challenge/asr2/local/score.sh:121]: thanks!
That is great.
But if it takes time, we can leave it for now.

Comment (simpleoier) [reply egs2/interspeech2024_dsu_challenge/asr2/local/score.sh:121]: I double-check it. Technically, it is almost done. As written in the README, the last python block can get the bitrate number as a variable. 
```bash
$ token_suffix="wavlm_large_21_km2000"
$ bpe_folder="data/token_list/src_bpe_unigram3000_rm_wavlm_large_21_km2000"
$ for dset in dev_clean dev_other test_clean test_other test_1h; do
$   result_dir="bitrate/${dset}"

$   paste \
$     <(<dump/raw/${dset}/text.rm.${token_suffix} cut -d" " -f1) \
$     <(<dump/raw/${dset}/text.rm.${token_suffix} spm_encode --model=${bpe_folder}/bpe.model --output_format=id) \
$     > dump/raw/${dset}/token_int.rm.${token_suffix}

$   python pyscripts/utils/convert_token2json.py \
$     --vocab data/token_list/src_bpe_unigram3000_rm_${token_suffix}/tokens.txt \
$     --token dump/raw/${dset}/token_int.rm.${token_suffix} \
$     --ref_scp data/${dset}/wav.scp \
$     --result_dir "${result_dir}"

$   python pyscripts/utils/calculate_bitrate.py \
$     --vocab ${result_dir}/vocab.json \
$     --tokens ${result_dir}/tokens.json \
$     --reference_len ${result_dir}/ref_len.scp \
$     --bitrate_details ${result_dir}/details.txt
$ done

$ python - <<EOF
import numpy as np
bitrates=[]
result_dir="${result_dir}/.."
for dset in ["dev_clean", "dev_other", "test_clean", "test_other", "test_1h"]:
    with open(f"{result_dir}/{dset}/details.txt", "r") as f:
        for line in f.readlines():
            lst = line.strip().split()
            bitrates.append(float(lst[1]))
print(np.mean(bitrates))

EOF
```

However, the only issue is the bpe folder name defined in asr2.sh: ```src_bpedir="${token_listdir}/src_bpe_${src_bpemode}${src_nbpe}_${src_case}_${src_lang}"```.
This is not directly passed to `local/score.sh`. If I add this argument, other recipes may have error.

Comment (sw005320) [reply egs2/interspeech2024_dsu_challenge/asr2/local/score.sh:121]: I see.
This is difficult...
@ftshijt, do you have some ideas?
If not, maybe we should give up...

File: egs2/interspeech2024_dsu_challenge/asr2/README.md (hunk 1 above)
Comment (simpleoier) [reply egs2/interspeech2024_dsu_challenge/asr2/README.md:9]: done.

File: egs2/interspeech2024_dsu_challenge/asr2/README.md [hunk 3]
```diff
@@ -24,27 +32,29 @@ $   python pyscripts/utils/convert_token2json.py \
...
-    with open(f"{result_dir}/{dset}/details.txt", "r") as f:
+    with open(f"{bitrate_dir}/{dset}/details.txt", "r") as f:
         for line in f.readlines():
             lst = line.strip().split()
             bitrates.append(float(lst[1]))
 print(np.mean(bitrates))
 
 EOF
+
+$ 16.399912599065225
```
Comment (sw005320) [inline egs2/interspeech2024_dsu_challenge/asr2/README.md:57]: Please round the value

Comment (simpleoier) [reply egs2/interspeech2024_dsu_challenge/asr2/README.md]: added round value to 2 decimal places

**Pull request merged**